overlord/servicestate/servicemgr.go

```diff
@@ -108,6 +108,31 @@ func (m *ServiceManager) ensureSnapServicesUpdated() (err error) {
 		return nil
 	}
 
+	for _, snapSt := range allStates {
+		info, err := snapSt.CurrentInfo()
+		if err != nil {
+			return err
+		}
+		squashfsPath := dirs.StripRootDir(info.MountFile())
+		whereDir := dirs.StripRootDir(info.MountDir())
+		var sysd systemd.Systemd
+		if snapdenv.Preseeding() {
+			sysd = systemd.NewEmulationMode(dirs.GlobalRootDir)
+		} else {
+			sysd = systemd.New(systemd.SystemMode, nil)
+		}
```

Comment:
This could be taken out of the loop (maybe behind a `len(allStates) > 0`, if it's expensive -- I don't know)

Reply:
Good point. It might be uselessly expensive on the garbage collection. I will move it.

Reply:
Done.